addons/web/static/src/views/relational_model/dynamic_list.js

```diff
@@ -74,6 +74,7 @@ export class DynamicList extends DataPoint {
         const context = this.context;
         const resIds = await this.getResIds(isSelected);
         const action = await this.model.orm.call(this.resModel, method, [resIds], { context });
+        this.count = this.count - resIds.length;
```

Comment:
this is wrong: you might have a filter allowing to display both active and inactive records
the updated count should come with the reload (also, it will mutate only once)